src/polygonId/verifier.ts

```diff
@@ -20,3 +24,111 @@ export default class Verifier {
     }
   }
 }
+
+// UTILS copied from:
+// https://github.com/0xPolygonID/contracts/blob/main/test/utils/deploy-utils.ts
+// when looking on how to construct params from submitZKPResponse
+function prepareInputs(zkp: ZeroKnowledgeProofResponse) {
+  const { proof, pub_signals } = zkp
+  const { pi_a, pi_b, pi_c } = proof
+  const [[p1, p2], [p3, p4]] = pi_b
+  const preparedProof = {
+    pi_a: pi_a.slice(0, 2),
+    pi_b: [
+      [p2, p1],
+      [p4, p3],
+    ],
+    pi_c: pi_c.slice(0, 2),
+  }
+
+  return { inputs: pub_signals, ...preparedProof }
+}
+
+export class OnChainVerifier {
+  private storage: StorageServices
+  private provider: Web3Provider
+  constructor(storage: StorageServices, provider: Web3Provider) {
+    this.storage = storage
+    this.provider = provider
+  }
+
+  async verify(zkp: ZeroKnowledgeProofResponse, zkpRequest: ZeroKnowledgeProofRequest): Promise<boolean> {
+    const network = await this.provider.getNetwork()
+    if (network.chainId !== 80001) {
+      // We support on-chain verification only on Mumbai network ATM
+      return true
+    }
+
+    try {
+      const validatorAddress = '0xA59B9E70639B2A4CF51af47f39D14B1E735301Fb'
```

Comment:
Link to the smart contracts/scripts that I'm using here: https://github.com/vacuumlabs/polygonid-tutorial-examples/pull/1/files